dev/io.openliberty.org.apache.myfaces.4.0_fat/fat/src/io/openliberty/org/apache/myfaces40/fat/tests/AnnotationLiteralsTest.java

```diff
@@ -0,0 +1,357 @@
+
+    private static final String APP_NAME = "AnnotationLiteralsTest";
+
+    @Server("faces40_AnnotationLiteralsTest")
+    public static LibertyServer server;
+
+    Map<String, String> testData = new HashMap<>();
+
+    @Rule
+    public TestName name = new TestName();
+
+    @BeforeClass
+    public static void setUp() throws Exception {
+        WebArchive app = ShrinkHelper.buildDefaultApp(APP_NAME + ".war", "io.openliberty.org.apache.faces40.fat.literals");
+
+//        //Test differences between myfaces and mojarra if needed
+//        app.addAsLibraries(new File("publish/files/mojarra40/").listFiles());
+//        app.addAsDirectories("publish/files/permissions");
+
+        ShrinkHelper.exportDropinAppToServer(server, app);
+        server.startServer(AnnotationLiteralsTest.class.getSimpleName() + ".log");
+    }
+
+    @AfterClass
+    public static void testCleanup() throws Exception {
+        // Stop the server
+        if (server != null && server.isStarted()) {
+            server.stopServer();
+        }
+    }
+
+    private URL getURL(String path, String... queries) throws Exception {
+        String finalQuery = "?" + name.getMethodName() + "=true";
+        for (String query : queries) {
+            finalQuery += "&" + query;
+        }
+        return HttpUtils.createURL(server, "/" + APP_NAME + "/" + path + finalQuery);
+    }
+
+    private HtmlPage getAndLogPage(WebClient webClient, String... queries) throws Exception {
+        return getAndLogPage("annotationLiterals.xhtml", webClient, queries);
+    }
+
+    private HtmlPage getAndLogPage(String path, WebClient webClient, String... queries) throws Exception {
+        URL url = getURL(path, queries);
+
+        HtmlPage page = (HtmlPage) webClient.getPage(url);
+
+        FATSuite.logOutputForDebugging(server, page.asXml(), name.getMethodName() + ".html");
+        return page;
+    }
+
+    /**
+     * Test data barAttribute set in TestAttributeFilter class of application
+     */
+    @Test
+    public void testApplicationMap() throws Exception {
+        String expected = "true";
+        try (WebClient webClient = new WebClient()) {
+            assertEquals("applicationMap did not produce an expected result", expected, getAndLogPage(webClient).getBody().asText());
+        }
+    }
+
+    /**
+     * Test data is the ClientBehavior class that uses the @FacesBehavior annotation
+     */
+    @Test
+    public void testFacesBehavior() throws Exception {
+        String expected = "true";
+        try (WebClient webClient = new WebClient()) {
+            assertEquals("facesBehavior did not produce the expected result", expected, getAndLogPage(webClient).getBody().asText());
+        }
+    }
+
+    /**
+     * Test data is the ConfigBean class that uses the @FacesConfig annotation
+     * FIXME - this doesn't currently work due to a NPE in the MyFaces impl
+     * Issue - https://github.com/jakartaee/faces/issues/1784
+     */
+//    @Test
```

Comment:
Just an FYI, I'll enable this test as part of : https://github.com/OpenLiberty/open-liberty/issues/24099